epochStart/bootstrap/fromLocalStorage.go

```diff
@@ -155,23 +157,54 @@ func (e *epochStartBootstrap) prepareEpochFromStorage() (Parameters, error) {
 func (e *epochStartBootstrap) checkIfShuffledOut(
 	pubKey []byte,
 	nodesConfig *sharding.NodesCoordinatorRegistry,
-) bool {
-	epochConfig := nodesConfig.EpochsConfig[fmt.Sprint(e.baseData.lastEpoch)]
-	shardIdForConfig := fmt.Sprint(e.baseData.shardId)
+) (uint32, bool) {
+	epochIDasString := fmt.Sprint(e.baseData.lastEpoch)
+	epochConfig := nodesConfig.EpochsConfig[epochIDasString]
 
-	for _, validator := range epochConfig.WaitingValidators[shardIdForConfig] {
-		if bytes.Equal(pubKey, validator.PubKey) {
-			return false
+	newShardId, isWaitingForShard := checkIfIsValidatorForEpoch(pubKey, epochConfig.WaitingValidators)
+	if isWaitingForShard {
+		isShuffledOut := newShardId != e.baseData.shardId
+		return newShardId, isShuffledOut
+	}
+
+	newShardId, isEligibleForShard := checkIfIsValidatorForEpoch(pubKey, epochConfig.WaitingValidators)
```

Comment:
duplicate call `checkIfIsValidatorForEpoch(pubKey, epochConfig.WaitingValidators)` something is wrong in the second call parameter

Reply:
thanks. resolved.